pkg/indexer/storer/groupMessage.go

```diff
@@ -2,23 +2,55 @@ package storer
 
 import (
 	"context"
-	"errors"
+	"fmt"
 
 	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/xmtp/xmtpd/pkg/abis"
 	"github.com/xmtp/xmtpd/pkg/db/queries"
 	"go.uber.org/zap"
 )
 
 type GroupMessageStorer struct {
-	queries *queries.Queries
-	logger  *zap.Logger
+	contract *abis.GroupMessages
+	queries  *queries.Queries
+	logger   *zap.Logger
 }
 
-func NewGroupMessageStorer(queries *queries.Queries, logger *zap.Logger) *GroupMessageStorer {
-	return &GroupMessageStorer{queries: queries, logger: logger}
+func NewGroupMessageStorer(
+	queries *queries.Queries,
+	logger *zap.Logger,
+	contract *abis.GroupMessages,
+) *GroupMessageStorer {
+	return &GroupMessageStorer{queries: queries, logger: logger, contract: contract}
 }
 
 // Validate and store a group message log event
 func (s *GroupMessageStorer) StoreLog(ctx context.Context, event types.Log) LogStorageError {
-	return NewLogStorageError(errors.New("not implemented"), true)
+	msgSent, err := s.contract.ParseMessageSent(event)
+	if err != nil {
+		return NewLogStorageError(err, false)
+	}
+
+	// TODO:nm figure out topic structure
+	topic := buildTopic(msgSent.GroupId)
+
+	s.logger.Debug("Inserting message from contract", zap.String("topic", topic))
+
+	if _, err = s.queries.InsertGatewayEnvelope(ctx, queries.InsertGatewayEnvelopeParams{
```

Comment:
Is it right that we don't ever expect to process the same log twice? If so, is it problematic if we run into a unique constraint error here (which would mean the (originator ID, sequence ID) tuple already exists)?

Reply:
We could if we have two instances of the replication node running, and both are indexing the chain.

If a node gets shut down mid-block, it'll also restart at the top of that block.

The only hard constraint is that they are both processing messages in ascending order.